<?xml version="1.0" encoding="UTF-8"?><testrun name="HibernateVehicleServiceTest.testVehicleServicegetVehiclesbyId" project="carshop-model" tests="1" started="1" failures="0" errors="1" ignored="0">
  <testcase name="testVehicleServicegetVehiclesbyId" classname="org.porourke.carshop.model.hibernate.HibernateVehicleServiceTest" time="0.199">
    <error>org.hibernate.exception.GenericJDBCException: Could not open connection&#13;
	at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:52)&#13;
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:125)&#13;
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:110)&#13;
	at org.hibernate.engine.jdbc.internal.LogicalConnectionImpl.obtainConnection(LogicalConnectionImpl.java:304)&#13;
	at org.hibernate.engine.jdbc.internal.LogicalConnectionImpl.getConnection(LogicalConnectionImpl.java:169)&#13;
	at org.hibernate.engine.jdbc.internal.proxy.ConnectionProxyHandler.extractPhysicalConnection(ConnectionProxyHandler.java:82)&#13;
	at org.hibernate.engine.jdbc.internal.proxy.ConnectionProxyHandler.continueInvocation(ConnectionProxyHandler.java:138)&#13;
	at org.hibernate.engine.jdbc.internal.proxy.AbstractProxyHandler.invoke(AbstractProxyHandler.java:81)&#13;
	at com.sun.proxy.$Proxy12.prepareStatement(Unknown Source)&#13;
	at org.hibernate.engine.jdbc.internal.StatementPreparerImpl$2.doPrepare(StatementPreparerImpl.java:105)&#13;
	at org.hibernate.engine.jdbc.internal.StatementPreparerImpl$StatementPreparationTemplate.prepareStatement(StatementPreparerImpl.java:166)&#13;
	at org.hibernate.engine.jdbc.internal.StatementPreparerImpl.prepareStatement(StatementPreparerImpl.java:103)&#13;
	at org.hibernate.id.insert.AbstractSelectingDelegate.performInsert(AbstractSelectingDelegate.java:52)&#13;
	at org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:2757)&#13;
	at org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:3268)&#13;
	at org.hibernate.action.internal.EntityIdentityInsertAction.execute(EntityIdentityInsertAction.java:78)&#13;
	at org.hibernate.engine.spi.ActionQueue.execute(ActionQueue.java:272)&#13;
	at org.hibernate.event.internal.AbstractSaveEventListener.performSaveOrReplicate(AbstractSaveEventListener.java:304)&#13;
	at org.hibernate.event.internal.AbstractSaveEventListener.performSave(AbstractSaveEventListener.java:195)&#13;
	at org.hibernate.event.internal.AbstractSaveEventListener.saveWithGeneratedId(AbstractSaveEventListener.java:128)&#13;
	at org.hibernate.event.internal.DefaultSaveOrUpdateEventListener.saveWithGeneratedOrRequestedId(DefaultSaveOrUpdateEventListener.java:204)&#13;
	at org.hibernate.event.internal.DefaultSaveEventListener.saveWithGeneratedOrRequestedId(DefaultSaveEventListener.java:55)&#13;
	at org.hibernate.event.internal.DefaultSaveOrUpdateEventListener.entityIsTransient(DefaultSaveOrUpdateEventListener.java:189)&#13;
	at org.hibernate.event.internal.DefaultSaveEventListener.performSaveOrUpdate(DefaultSaveEventListener.java:49)&#13;
	at org.hibernate.event.internal.DefaultSaveOrUpdateEventListener.onSaveOrUpdate(DefaultSaveOrUpdateEventListener.java:90)&#13;
	at org.hibernate.internal.SessionImpl.fireSave(SessionImpl.java:642)&#13;
	at org.hibernate.internal.SessionImpl.save(SessionImpl.java:635)&#13;
	at org.hibernate.internal.SessionImpl.save(SessionImpl.java:631)&#13;
	at org.porourke.carshop.model.hibernate.VehicleService.saveVehicle(VehicleService.java:55)&#13;
	at org.porourke.carshop.model.hibernate.HibernateVehicleServiceTest.testVehicleServicegetVehiclesbyId(HibernateVehicleServiceTest.java:91)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)&#13;
	at java.lang.reflect.Method.invoke(Method.java:606)&#13;
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)&#13;
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)&#13;
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)&#13;
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)&#13;
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)&#13;
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)&#13;
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)&#13;
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)&#13;
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)&#13;
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)&#13;
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)&#13;
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)&#13;
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)&#13;
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)&#13;
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:467)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:683)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:390)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:197)&#13;
Caused by: org.h2.jdbc.JdbcSQLException: Database may be already in use: "C:/Projects/workspace/FirstProject/javabrains/hibernate/test.mv.db". Possible solutions: close all other connection(s); use the server mode [90020-185]&#13;
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:345)&#13;
	at org.h2.message.DbException.get(DbException.java:168)&#13;
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:108)&#13;
	at org.h2.engine.Database.getPageStore(Database.java:2379)&#13;
	at org.h2.engine.Database.open(Database.java:666)&#13;
	at org.h2.engine.Database.openDatabase(Database.java:266)&#13;
	at org.h2.engine.Database.&lt;init&gt;(Database.java:260)&#13;
	at org.h2.engine.Engine.openSession(Engine.java:60)&#13;
	at org.h2.engine.Engine.openSession(Engine.java:167)&#13;
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:145)&#13;
	at org.h2.engine.Engine.createSession(Engine.java:128)&#13;
	at org.h2.engine.Engine.createSession(Engine.java:26)&#13;
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:347)&#13;
	at org.h2.jdbc.JdbcConnection.&lt;init&gt;(JdbcConnection.java:108)&#13;
	at org.h2.jdbc.JdbcConnection.&lt;init&gt;(JdbcConnection.java:92)&#13;
	at org.h2.Driver.connect(Driver.java:72)&#13;
	at java.sql.DriverManager.getConnection(DriverManager.java:571)&#13;
	at java.sql.DriverManager.getConnection(DriverManager.java:187)&#13;
	at org.hibernate.service.jdbc.connections.internal.DriverManagerConnectionProviderImpl.getConnection(DriverManagerConnectionProviderImpl.java:173)&#13;
	at org.hibernate.internal.AbstractSessionImpl$NonContextualJdbcConnectionAccess.obtainConnection(AbstractSessionImpl.java:276)&#13;
	at org.hibernate.engine.jdbc.internal.LogicalConnectionImpl.obtainConnection(LogicalConnectionImpl.java:297)&#13;
	... 51 more&#13;
Caused by: java.lang.IllegalStateException: The file is locked: nio:C:/Projects/workspace/FirstProject/javabrains/hibernate/test.mv.db [1.4.185/7]&#13;
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:773)&#13;
	at org.h2.mvstore.FileStore.open(FileStore.java:170)&#13;
	at org.h2.mvstore.MVStore.&lt;init&gt;(MVStore.java:346)&#13;
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2769)&#13;
	at org.h2.mvstore.db.MVTableEngine$Store.&lt;init&gt;(MVTableEngine.java:162)&#13;
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:98)&#13;
	... 69 more&#13;
&#13;
</error>
  </testcase>
</testrun>
